### ISSUE TITLE
Request for clarification on the architecture

### ISSUE THREAD

**Issue by yerramasu** [Ziyan-Huang/STU-Net#16]
Hello Ziyan,

Thanks you very much for the repo. 

I am beginner and I am trying to understand Continous learning models. 

I have a few question, I have read that in Continuous Deep Learning models there is an issue with Catastrophic Forgetfulness. How are you able to over come this issue. I apologize if my question is naive.

Also I was not able to download weights for FLARE23 from Baidu, could you please upload them to gdrive. I would like to try training on TCIA NSCLC dataset.

Regards,
Anil

**Comment by Ziyan-Huang** [comment]
Hi Anil,

In a typical Continuous Learning setup, models are trained sequentially on different datasets, with the objective of retaining performance across all datasets throughout this process. 

However, STU-Net follows a more traditional pretraining-finetuning approach. Initially, a pre-training phase is conducted on a large-scale dataset, after which the model is fine-tuned for different downstream tasks to improve performance on these tasks. It's important to mention that Catastrophic Forgetfulness can occur during this process, leading to a potential loss in performance on the upstream tasks. This issue has not been addressed in my setup as it's not within my current concerns.

Nonetheless, I believe this is a very worthwhile issue to investigate.

I will make sure to upload the weights of FLARE23 to Google Drive

Best,

Ziyan

**Comment by yerramasu** [comment]
Hello Ziyan,

Thank you very much for taking time to explain my doubt.

I will work on TCIA NSCLC dataset for lung nodule detection. I will update you with the results after I complete the training 

Kind Regards,
Anil

**Comment by Airliin** [comment]
Hello, how to use run_finetuning.py for downstream task fine-tuning, still can maintain the original category capabilities; For example, I use run_finetuning.py in a downstream task, which only has three classes. If I train it directly, the network will only split these three classes.

**Comment by Ziyan-Huang** [comment]
Hello @Airliin . The current run_finetuning.py does not support maintaining the original category capabilities, and I can't provide much help in this area. However, you can refer to some continual learning papers for guidance.

**Comment by yerramasu** [comment]
@Ziyan-Huang . I am working on porting your model to nnUNet v2. Can I do a pull request after the update. I feel it would be better for others to have a single repo instead of having a separate repo :)

Regards,
Anil

**Comment by Airliin** [comment]
Hello @Ziyan-Huang  .Thank you for your response. I noticed in your article you mentioned reducing the learning rate of the pre-trained weights to less than ten times that of 'seghead' when running 'run_finetuning.py', but I didn't see this setting in the code. Could you please advise on where this can be modified?

**Comment by Ziyan-Huang** [comment]
Dear Anil

@yerramasu, Thank you for your initiative! We truly appreciate your effort. We have indeed provided a basic version of the nnUNet v2 implementation at https://github.com/Ziyan-Huang/STU-Net/tree/main/nnUNet-2.2. If you find any areas of improvement or optimization, we would be more than happy to review and potentially incorporate your changes. Pull requests are very much welcome.

Best regards,

Ziyan Huang

**Comment by Ziyan-Huang** [comment]
Dear @Airlin,

Thank you for pointing that out. Currently, our code does not explicitly provide this setting. Due to your suggestion, we are considering incorporating this feature. However, to be honest, using a uniformly adjusted learning rate yields similar results to the differentiated setting you mentioned for the 'seghead'. 

Best regards,

Ziyan Huang

**Comment by yerramasu** [comment]
Hello @Ziyan-Huang , thank you very much. I wilose this issue as my query has been addressed and open Issue tracking for nnUNet v2.

Thanks again for the great repo.

Regards,
Anil
